cucumber/authenticators_jwt/features/authn_jwt_token_schema.feature

```diff
@@ -448,4 +450,266 @@ Feature: JWT Authenticator - Token Schema
+      member: !host myapp
+    """
+    And I successfully set authn-jwt "mapping-claims" variable to value "sub:ref"
+    And I issue a JWT token:
+    """
+    {
+      "host":"myapp",
+      "sub":"mysub",
+      "ref":"mybranch"
+    }
+    """
+    And I save my place in the log file
+    When I authenticate via authn-jwt with the JWT token
+    Then the HTTP response status code is 401
+    And The following appears in the log after my savepoint:
+    """
+    CONJ00049E Resource restriction 'sub' does not match with the corresponding value in the request
+    """
+
+  Scenario: ONYX-10860 - Mapping claims configured but not populated - 401 Error
+    Given I extend the policy with:
+    """
+    - !variable conjur/authn-jwt/raw/mapping-claims
+    - !variable conjur/authn-jwt/raw/enforced-claims
+
+    - !host
+      id: myapp
+      annotations:
+        authn-jwt/raw/sub: valid
+
+    - !grant
+      role: !group conjur/authn-jwt/raw/hosts
+      member: !host myapp
+    """
+    And I issue a JWT token:
+    """
+    {
+      "ref":"valid",
+      "host":"myapp"
+    }
+    """
+    And I successfully set authn-jwt "enforced-claims" variable to value "ref"
+    And I save my place in the log file
+    When I authenticate via authn-jwt with the JWT token
+    Then the HTTP response status code is 401
+    And The following appears in the log after my savepoint:
+    """
+     CONJ00037E Missing value for resource: cucumber:variable:conjur/authn-jwt/raw/mapping-claims
+    """
+
+  @sanity
+  Scenario: ONYX-11117: Enforced Claims and Mappings with special allowed characters. Annotations are correct. 200 OK
+    Given I extend the policy with:
+    """
+    - !variable conjur/authn-jwt/raw/mapping-claims
+    - !variable conjur/authn-jwt/raw/enforced-claims
+
+    - !host
+      id: myapp
+      annotations:
+        authn-jwt/raw/claim.name: claim.name.value  # Only Enforce
+        authn-jwt/raw/claim_ant: claim.ant...value # Map And Enforce
+        authn-jwt/raw/_: claim_name_value # Only Map
+
+    - !grant
+      role: !group conjur/authn-jwt/raw/hosts
+      member: !host myapp
+    """
+    And I successfully set authn-jwt "mapping-claims" variable to value "claim_ant:claim.ant..., _:claim_name"
+    And I successfully set authn-jwt "enforced-claims" variable to value "claim.name, claim.ant..."
+    And I issue a JWT token:
+    """
+    {
+      "host":"myapp",
+      "claim.name": "claim.name.value",
+      "claim.ant...": "claim.ant...value",
+      "claim_name": "claim_name_value"
+    }
+    """
+    And I save my place in the audit log file
+    When I authenticate via authn-jwt with the JWT token
+    Then the HTTP response status code is 200
+    And The following appears in the log after my savepoint:
+    """
+    cucumber:host:myapp successfully authenticated with authenticator authn-jwt service cucumber:webservice:conjur/authn-jwt/raw
+    """
+
+  Scenario Outline: ONYX-10873 - Broken claims mapping - 401 Error
```

Comment:
what does `scenario outline` means?

Reply:
https://stackoverflow.com/questions/43497249/how-to-define-steps-for-scenario-outline-with-ruby